lib/class-wp-rest-menu-items-controller.php

```diff
@@ -251,6 +275,16 @@ protected function prepare_item_for_database( $request ) {
 			}
 		}
 
+		$taxonomy = get_taxonomy( 'nav_menu' );
+		$base     = ! empty( $taxonomy->rest_base ) ? $taxonomy->rest_base : $taxonomy->name;
+		// If menus submitted, then check if 1 menu is given, if more than 1 is given, error out.
+		if ( isset( $request[ $base ] ) && ! empty( $request[ $base ] ) ) {
```

Comment:
What this is checking exactly? Sorry, not following it completely.

Reply:
Check if `$request['menu']`is set and not empty. Menu items can't be assigned to multiple menu items.

Reply:
I think we should be able to express this in the schema. Going to take a look later.

Reply:
How would this be expressed in the schema?

Reply:
You can do that using `maxItems` but we don't currently support that in `rest_validate_value_from_schema`.

However, if a menu item can only be assigned to one menu, can we make the property an integer instead of an array of 1 integer?

Reply:
@TimothyBJacobs Any chance you could provide a code example to explain what you mean here?

Reply:
The schema definition for using `maxItems` would be like this.

```
type: 'array',
items: [ type: 'integer' ],
maxItems: 1
```

> However, if a menu item can only be assigned to one menu, can we make the property an integer instead of an array of 1 integer?

So we'd then have.

```
type: 'integer'
```

Reply:
So to confirm, `maxItems` is not currently supported. This is a new field? I am not against this, it is cool. Just confirming.

Reply:
Yeah, we'd have to add it to `rest_validate_value_from_schema`. I'd like to add that regardless. But my preferred solution would be to just make the `menu` property a single integer.

Reply:
Fixed in https://github.com/WP-API/menus-endpoints/pull/31/commits/1cec7b9519a711ff295778c726f99f6538a4930f

Reply:
> But my preferred solution would be to just make the menu property a single integer.

Is this not doable for some reason?

Reply:
@TimothyBJacobs How about this https://github.com/WP-API/menus-endpoints/pull/31/commits/d0d25bfa92f22046943c346cf81e603a9afe7b56

Reply:
That looks good to me. Is it intentional that you can't remove a menu item from a menu?

Reply:
You can only resign a menu item, not have it floating with a menu. IMO.

Reply:
That works for me.